lib/trino-parquet/src/main/java/io/trino/parquet/ParquetCompressionUtils.java

```diff
@@ -70,34 +70,42 @@ public static Slice decompress(CompressionCodec codec, Slice input, int uncompre
         throw new ParquetCorruptionException("Codec not supported in Parquet: " + codec);
     }
 
-    private static Slice decompressSnappy(Slice input, int uncompressedSize)
+    public interface OutputBuffer
+    {
+        byte[] initialize(int size);
```

Comment:
Shouldn't it be more like `aquire/release` semantics? When buffer is released? Add a comment please

Reply:
Buffer was tied to PageReader which is tied to per-row group column reader which gets cleaned up whenever we reach of end of a row group

Reply:
Sematics here is similar to the OutputBuffer in orc reader